NOTE: keep this patch in sync between zeroc-ice35 and py-zeroc-ice35.

Changes:

1) Allow environment to set CXX.
2) Do not build for both i386 and x86_64 since MacPorts' universal
   configuration will specify which architectures to build for.
3) Do not define -DMAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_6
   so that any Mac OS X version can be used.  Since we link against
   MacPorts' OpenSSL, this flag is unnecessary.

diff -ru Ice-3.5.1.orig/cpp/config/Make.rules.Darwin Ice-3.5.1/cpp/config/Make.rules.Darwin
--- Ice-3.5.1.orig/cpp/config/Make.rules.Darwin	2013-10-04 08:48:14.000000000 -0700
+++ Ice-3.5.1/cpp/config/Make.rules.Darwin	2013-10-09 13:53:53.882488377 -0700
@@ -11,18 +11,12 @@
 # This file is included by Make.rules when uname is Darwin.
 #
 
-CXX			= xcrun clang++
+CXX			?= xcrun clang++
 
 CPPFLAGS 	        += -pthread
 CXXFLAGS		+= -Wall -Werror
 
 ifeq ($(OPTIMIZE),yes)
-     #
-     # By default we build binaries with both architectures when optimization is enabled.
-     #
-     ifeq ($(CXXARCHFLAGS),)
-     	CXXARCHFLAGS	:= -arch i386 -arch x86_64
-     endif   
      CXXFLAGS		:= $(CXXARCHFLAGS) -O2 -DNDEBUG $(CXXFLAGS)
 else
      CXXFLAGS		:= $(CXXARCHFLAGS) -g $(CXXFLAGS)
@@ -66,11 +60,6 @@
 
 mkshlib     = $(CXX)  -dynamiclib $(shlibldflags) -o $(1) -install_name @rpath/$(2) $(3) $(4)
 
-#
-# Avoid OpenSSL deprecation warnings on OS X
-#
-OPENSSL_FLAGS   	= -DMAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_6
-
 BASELIBS		= -lIceUtil 
 LIBS			= -lIce $(BASELIBS)
 
